src/applications/mhv-medical-records/reducers/vitals.js

```diff
@@ -101,7 +102,21 @@ export const extractLocation = vital => {
 };
 
 export const convertVital = record => {
-  const type = macroCase(record.code?.text);
+  // Determine canonical vital type via any mapped LOINC present in code.coding
+  let type;
+  if (isArrayAndHasItems(record.code?.coding)) {
+    for (const coding of record.code.coding) {
+      if (loincToVitalType[coding.code]) {
+        type = loincToVitalType[coding.code];
+        break;
+      }
+    }
+  }
+  // Fallback: derive from text (legacy) else mark as OTHER
+  if (!type) {
+    const derived = macroCase(record.code?.text);
+    type = loincToVitalType[derived] || derived || 'OTHER';
```

Comment:
TODO: add some logging / monitoring here so we can track what is getting missed and update our LOINC list if we need to.

Reply:
added to do, will address in future PR.